File: commonCharacters.js
var commonChars = function(words) {
    common_counts = Array(26).fill(0);
    curr_counts =  Array(26).fill(0);
    
    for (let c of words[0]) {
        common_counts[c.charCodeAt(0)-'a'.charCodeAt(0)]++;
    }

    for (let i = 1; i < words.length; i++) {
        let currStr = words[i];
        curr_counts.fill(0);
        for (let c of currStr) {
            curr_counts[c.charCodeAt(0)-'a'.charCodeAt(0)]++;
        }
        for (let j = 0; j < 26; j++) {
            common_counts[j] = Math.min(common_counts[j], curr_counts[j])
        }
    }

    let ans = [];
    for (let i = 0; i < 26; i++) {
        let repeat = common_counts[i];
        for (let j = 0; j < repeat; j++) {
            ans.push(String.fromCharCode(i + 'a'.charCodeAt(0)));
        }
    }

    return ans;
};

console.log(commonChars(["bella","label","roller"]));

File: commonCharacters.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class commonCharacters {
    public List<String> commonChars(String[] words) {
        int[] commonCounts = new int[26];
        int[] currCounts = new int[26];

        for (int i = 0; i < words[0].length(); i++) {
            commonCounts[words[0].charAt(i)-'a']++;
        }

        for (int i = 1; i < words.length; i++) {
            String currString = words[i];
            Arrays.fill(currCounts, 0);
            for (int j = 0; j < currString.length(); j++) {
                currCounts[currString.charAt(j)-'a']++;
            }
            for (int j = 0; j < 26; j++) {
                commonCounts[j] = Math.min(commonCounts[j], currCounts[j]);
            }
        }

        List<String> ans = new ArrayList();
        for (int j = 0; j < 26; j++) {
            int repeat = commonCounts[j];
            for (int i = 0; i < repeat; i++) {
                ans.add(String.valueOf((char) ('a' + j)));
            }
        }

        return ans;
    }

    public static void main(String[] args) {
        String[] words = new String[]{"bella", "label", "roller"};
        commonCharacters commonCharacters = new commonCharacters();
        System.out.println(commonCharacters.commonChars(words));
    }
}
